plugins/woocommerce/includes/class-wc-template-loader.php

```diff
@@ -107,9 +107,17 @@ private static function has_block_template( $template_name ) {
 			return false;
 		}
 
-		return is_readable(
-			get_stylesheet_directory() . '/block-templates/' . $template_name . '.html'
-		);
+		$has_template = (bool) is_readable( get_stylesheet_directory() . '/block-templates/' . $template_name . '.html' );
+
+		/**
+		 * Filters the value of the result of the block template check.
+		 *
+		 * @since x.x.x
+		 *
+		 * @param boolean $has_template value to be filtered.
+		 * @param string $template_name The name of the template.
+		 */
+		return apply_filters( 'woocommerce_has_block_template', $has_template, $template_name );
```

Comment:
Hi! My earlier comment probably wasn't clear enough, but this is what I meant re applying a cast (basically, apply it to the `apply_filters()` call before we return):

```suggestion
		$has_template = is_readable( get_stylesheet_directory() . '/block-templates/' . $template_name . '.html' );

		/**
		 * Filters the value of the result of the block template check.
		 *
		 * @since x.x.x
		 *
		 * @param boolean $has_template value to be filtered.
		 * @param string $template_name The name of the template.
		 */
		return (bool) apply_filters( 'woocommerce_has_block_template', $has_template, $template_name );
```

To explain: the [`is_readable()`](https://www.php.net/manual/en/function.is-readable.php) function always returns a boolean, so no cast needed there. However, this filter can be used by anyone, including 3PDs, and they may unintentionally change the type during filtering. So, the cast is here to guarantee the declared return type per the method's docblock.

Reply:
Ah apologies @barryhughes . Makes total sense, not sure how I misinterpreted that looking back. I've committed you suggestion. Thank you.